sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,1030 @@
+     *        items.
+     */
+    int32_t MaxRecords = int32_t();
+
+    /**
+     * @brief Sets POSIX access control rights on files and directories. The value is a
+     *        comma-separated list of access control entries. Each access control entry (ACE)
+     *        consists of a scope, a type, a user or group identifier, and permissions in the format
+     *        "[scope:][type]:[id]:[permissions]".
+     */
+    std::string Acl;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::SetProperties
+   */
+  struct SetPathPropertiesOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief User-defined metadata to be stored with the filesystem.
+     *        Note that the string may only contain ASCII characters in the
+     *        ISO-8859-1 character set.
+     */
+    std::map<std::string, std::string> Metadata;
+
+    /**
+     * @brief Sets the blob's cache control. If specified, this property is stored with the
+     *        blob and returned with a read request.
+     */
+    std::string CacheControl;
+
+    /**
+     * @brief Sets the blob's content type. If specified, this property is stored
+     *        with the blob and returned with a read request.
+     */
+    std::string ContentType;
+
+    /**
+     * @brief Sets the blob's Content-Disposition header.
+     */
+    std::string ContentDisposition;
+
+    /**
+     * @brief Sets the blob's content encoding. If specified, this property is stored with
+     *        the blob and returned with a read request.
+     */
+    std::string ContentEncoding;
+
+    /**
+     * @brief Set the blob's content language. If specified, this property is stored with
+     *        the blob and returned with a read request.
+     */
+    std::string ContentLanguage;
+
+    /**
+     * @brief Specify an ETag value to operate only on path with a matching value.
+     */
+    std::string IfMatch;
+
+    /**
+     * @brief Specify an ETag value to operate only on path without a matching value.
+     */
+    std::string IfNoneMatch;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has been modified since the
+     (        specified date and time.
+    */
+    std::string IfModifiedSince;
+
+    /**
+     * @brief A date and time value. Specify this header to perform the
+     *        operation only if the resource has not been modified since
+     *        the specified date and time.
+     */
+    std::string IfUnmodifiedSince;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakePathClient::Update
+   * @remark Some optional parameter is mandatory when certain action is specified for the update
+   *         operation. More details:
+   * https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update
+   */
+  struct PathUpdateOptions : public SharedRequestOptions
```

Comment:
You put options of four actions into one struct. This misleads me to think I can flush data and set properties and set ACL for a file in one shot.

Reply:
The documentation clearly states that some options are bound to certain actions. Customer should not be using this API unless this is not clear to them. See the remarks:
```
     * @remark This API is not advised to be used unless absolutely needed. Please use AppendData,
     *         FlushData, SetAccessControl, SetAccessControlRecursive, SetProperties instead for
     *         specific functionality.
```